pkg/util/cancel_monitor.go

```diff
@@ -22,11 +22,10 @@ import (
 )
 
 // MonitorCancelLatency monitors the latency from ctx being cancelled and the returned function being called
-func MonitorCancelLatency(ctx context.Context, identifier string) func() {
+func MonitorCancelLatency(ctx context.Context, identifier string) (func(), func()) {
```

Comment:
Could you add some comments about the returned closures?

Reply:
done